src/nspawn/nspawn.c

```diff
@@ -186,6 +186,7 @@ static char *arg_image = NULL;
 static char *arg_oci_bundle = NULL;
 static VolatileMode arg_volatile_mode = VOLATILE_NO;
 static ExposePort *arg_expose_ports = NULL;
+static FirewallContext *fw_ctx = NULL;
```

Comment:
I will wait here until the 'do we want explicit initialization or not' clears up. If implicit/lazy is the way to go then different context could be used as well, as the end of run only has to 'clean up exposed ports', i.e. this only occurs once. Doing an extra
context init for that doesn't look like a problem and it would avoid having to pass the context from run up the call chain.